### PR TITLE
test(ddtrace/x/agenttest): make the mock agent's /info response configurable

### PR DESCRIPTION
**Pull request by darccio**
## What does this PR do?

First in a split of [#5122](https://github.com/DataDog/dd-trace-go/pull/5122) into a reviewable stack of smaller PRs (see that PR for the full sequence and rationale). This PR is standalone against `main`.

The mock agent always advertised exactly the patterns registered via `HandleTraces`, which left tests unable to express "the agent serves this endpoint but does not advertise it" — the shape agent-capability negotiation actually has to cope with.

Adds `SetInfoEndpoints` to decouple what `/info` advertises from what is registered, and uses it to pin the `ddtrace/tracer` inspectable-tracer harness to v0.4 on purpose. Those suites already run on v0.4, but only as a side effect: the mock advertises no `/v0.6/stats`, so client-side stats are unavailable and the tracer's protocol gate downgrades v1.0 -> v0.4. Making the pin explicit is a no-op today and keeps the harness on a deterministic wire format the next time that gate is touched (removed later in this stack).

Also fixes `handleV1Traces`, found while testing this stack: it built a bare `&payloadV1{buf: body}`, leaving `header` nil, so `decodeBuffer` panicked with an index-out-of-range inside `updateHeader`. The handler had been registered but unreachable since it was written, because the gate always selected v0.4 for this mock, so the bug never surfaced. `TestHandleV1TracesDecodesEncodedPayload` reproduces the panic without the fix.

Also guards the `endpoints` slice with the existing mutex on both sides — `HandleTraces` appended to it unlocked while `handleInfo` read it.

### Reviewer's Checklist

- [x] Changed code has unit tests for its functionality at or near 100% coverage.
- [x] New code doesn't break existing tests. You can check this by running `make test` locally.
- [x] New code is free of linting errors. You can check this by running `make lint` locally.
- [x] All generated files are up to date. You can check this by running `make generate` locally.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Review by kakkoyun — APPROVED** [review]

File: ddtrace/x/agenttest/agent.go [hunk 1]
```diff
@@ -123,8 +146,14 @@ func (a *agent) HandleTraces(pattern string, handler TraceHandler) {
...
 func (a *agent) handleInfo(w http.ResponseWriter, _ *http.Request) {
+	a.mu.Lock()
+	endpoints := a.endpoints
+	if a.infoEndpoints != nil {
+		endpoints = a.infoEndpoints
+	}
+	a.mu.Unlock()
 	w.Header().Set("Content-Type", "application/json")
-	json.NewEncoder(w).Encode(map[string]any{"endpoints": a.endpoints, "client_drop_p0s": true})
+	json.NewEncoder(w).Encode(map[string]any{"endpoints": endpoints, "client_drop_p0s": true})
```
[the remark this answers is not in the document]
Comment (darccio) [reply ddtrace/x/agenttest/agent.go:164]: Fixed in 14b24ecf2 — registered a no-op `/v0.6/stats` handler on `New()` so advertising it via `SetInfoEndpoints` is always backed by a reachable route, plus a regression test (`TestAgent_StatsEndpoint_AdvertisedIsReachable`) covering exactly this scenario.
